File: src-jvm/main/java/org/develnext/jphp/ext/controlsfx/classes/control/UXRangeSlider.java
package org.develnext.jphp.ext.controlsfx.classes.control;

import javafx.geometry.Orientation;
import javafx.scene.Node;
import javafx.util.Duration;
import org.controlsfx.control.PopOver;
import org.controlsfx.control.RangeSlider;
import org.develnext.jphp.ext.controlsfx.ControlsFXExtension;
import org.develnext.jphp.ext.javafx.classes.UXControl;
import php.runtime.annotation.Reflection;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Namespace(ControlsFXExtension.NS)
@Reflection.Name("UXRangeSlider")
public class UXRangeSlider extends UXControl<RangeSlider> {
    public UXRangeSlider(Environment env, RangeSlider wrappedObject) {
        super(env, wrappedObject);
    }
    public UXRangeSlider(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }
    interface WrappedInterface{
        @Reflection.Property boolean highValueChanging();
        @Reflection.Property boolean lowValueChanging();
        @Reflection.Property boolean showTickLabels();
        @Reflection.Property boolean showTickMarks();
        @Reflection.Property boolean snapToTicks();

        @Reflection.Property double blockIncrement();
        @Reflection.Property double highValue();
        @Reflection.Property double lowValue();
        @Reflection.Property double majorTickUnit();
        @Reflection.Property double max();
        @Reflection.Property double min();

        @Reflection.Property int minorTickCount();

        @Reflection.Property Orientation orientation();
    }

    @Reflection.Signature
    public void __construct(){
        __wrappedObject = new RangeSlider();
    }
}

File: sdk/php/gui/controlsfx/UXRangeSlider.php
<?php
namespace php\gui\controlsfx;

use php\gui\UXControl;

class UXRangeSlider extends UXControl{
    /**
     * @var double
    */
    public $min;
    /**
     * @var double
    */
    public $max;
    /**
     * @var string
     * HORIZONTAL, VERTICAL
    */
    public $orientation;
    /**
     * @var boolean
    */
    public $showTickLabels;
    /**
     * @var boolean
    */
    public $showTickMarks;
    /**
     * @var double
    */
    public $majorTickUnit;
    /**
     * @var int
    */
    public $minorTickCount;
    /**
     * @var boolean
    */
    public $snapToTicks;
    /**
     * @var double
    */
    public $blockIncrement;
    /**
     * @var double
    */
    public $lowValue;
    /**
     * @var boolean
    */
    public $lowValueChanging;
    /**
     * @var double
    */
    public $highValue;
    /**
     * @var boolean
    */
    public $highValueChanging;

}